ramlfications/parser/types.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2015 Spotify AB
+
+from __future__ import absolute_import, division, print_function
+import attr
+import copy
+from six import MAXSIZE
+from six import iteritems
+
+from ..errors import UnknownTypeError
+from ..parameters import Content
+
+__type_registry = {}
+
+def type_class(klass):
+    if klass.type is not None:
+        __type_registry[klass.type] = klass
+    return klass
```

Comment:
its the registry design pattern which is used in twisted/zope interfaces in a couple of places.